nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DistributeHashRecord.java

```diff
@@ -0,0 +1,401 @@
+@DynamicProperty(name = "The name of the relationship to route data to",
+        value = "Weight for this relationship.",
+        expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
+        description = "Each user-defined property specifies a relationship and weight for this.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "mime.type", description = "Sets the mime.type attribute to the MIME Type specified by the Record Writer"),
+        @WritesAttribute(attribute = "record.count", description = "The number of records selected by the query")
+})
+public class DistributeHashRecord extends AbstractProcessor {
+
+    public static final String MURMURHASH_32 = "murmurhash_32";
+
+    public static PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+            .name("reader")
+            .displayName("Record Reader")
+            .identifiesControllerService(RecordReaderFactory.class)
+            .required(true)
+            .description("Specifies the Controller Service to use for parsing incoming data and determining the data's schema")
+            .build();
+
+    public static PropertyDescriptor RECORD_WRITER = new PropertyDescriptor.Builder()
+            .name("writer")
+            .displayName("Record Writer")
+            .identifiesControllerService(RecordSetWriterFactory.class)
+            .required(true)
+            .description("Specifies the Controller Service to use for writing out the records")
+            .build();
+
+    private static final String KEY_DELIMITER = ",";
+
+    public static PropertyDescriptor KEYS = new PropertyDescriptor.Builder()
+            .name("keys")
+            .displayName("Keys")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .description("Field names in a record separated by commas." +
+                    " If record has one key and this key is integer or long then hash function " +
+                    "will not be evaluated and processor will distribute record by this numerical value. " +
+                    "If this record has several keys for distribution or one key with not 'int' or 'long' type then processor will " +
+                    "obtain keys from record, trim them and join with `-` delimiter like <firstKey>-<secondKey>-<...> " +
+                    "then evaluate hash function which return numerical value for distribution")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static PropertyDescriptor HASH_FUNCTION = new PropertyDescriptor.Builder()
+            .name("hash function")
+            .displayName("Hash Function")
+            .required(true)
+            .description("Hash algorithm for keys hashing")
+            .allowableValues(MURMURHASH_32)
```

Comment:
I'm curious what this hash function is, and why there is a configurable property for it if there is only one allowable value.

Reply:
This is name of hash function for distribution. I suggest there will be several hash functions in future (MD5 for example). After keys hashing processor will decide where record should route (by division without remainder)

Reply:
Is it possible to add the other hashing algorithms now?

Reply:
Yes! Done!